### PR TITLE
Improve browser experience

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses 
[#458](https://github.com/MetaMask/mobile-planning/issues/458)
[#354](https://github.com/MetaMask/mobile-planning/issues/354)

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: patches/react-native-webview+11.13.0.patch [hunk 1]
```diff
@@ -204,67 +210,144 @@ index f743bbc..f4e9b62 100644
...
-+                }
-+              }
+     webView.setDownloadListener(new DownloadListener() {
++      protected ReactContext mReactContext;
++
+       public void onDownloadStart(String url, String userAgent, String contentDisposition, String mimetype, long contentLength) {
+         webView.setIgnoreErrFailedForThisURL(url);
++        this.mReactContext = reactContext;
+ 
++        ArrayList<String> requestedAndroidPermissions = new ArrayList<>();
```
Comment (Cal-L) [inline patches/react-native-webview+11.13.0.patch:231]: Doesn't look like this is being used.

File: patches/react-native-webview+11.13.0.patch [hunk 2]
```diff
@@ -204,67 +210,144 @@ index f743bbc..f4e9b62 100644
...
++        ArrayList<String> grantedPermissions = new ArrayList<>();
```
Comment (Cal-L) [inline patches/react-native-webview+11.13.0.patch:232]: Doesn't look like this is being used.

File: patches/react-native-webview+11.13.0.patch [hunk 3]
```diff
@@ -476,20 +576,9 @@ index f743bbc..f4e9b62 100644
  
      public void setIgnoreErrFailedForThisURL(@Nullable String url) {
        ignoreErrFailedForThisURL = url;
-@@ -857,24 +1104,27 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
-     @Override
-     public void onPageFinished(WebView webView, String url) {
-       super.onPageFinished(webView, url);
-+      // Only return the URL that the web view is currently showing.
-+      String visibleUrl = webView.getUrl();
```
Comment (Cal-L) [inline patches/react-native-webview+11.13.0.patch:484]: This shouldn't be removed. Can you verify your merge from main to confirm that you have the latest changes?

File: patches/react-native-webview+11.13.0.patch [hunk 4]
```diff
@@ -584,16 +673,255 @@ index f743bbc..f4e9b62 100644
...
 +    protected boolean blockJsDuringLoading = true; //This boolean block JS prompts and alerts from displaying during loading
  
      /*
       * - Permissions -
-@@ -1402,6 +1696,15 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+@@ -1217,19 +1528,40 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+       ArrayList<String> requestedAndroidPermissions = new ArrayList<>();
+       for (String requestedResource : request.getResources()) {
+         String androidPermission = null;
++        String requestPermissionIdentifier = null;
```
Comment (Cal-L) [inline patches/react-native-webview+11.13.0.patch:872]: Just tagging this line as actual code that changed

File: patches/react-native-webview+11.13.0.patch [hunk 5]
```diff
@@ -204,67 +210,144 @@ index f743bbc..f4e9b62 100644
...
+       public void onDownloadStart(String url, String userAgent, String contentDisposition, String mimetype, long contentLength) {
+         webView.setIgnoreErrFailedForThisURL(url);
++        this.mReactContext = reactContext;
+ 
++        ArrayList<String> requestedAndroidPermissions = new ArrayList<>();
++        ArrayList<String> grantedPermissions = new ArrayList<>();
+         RNCWebViewModule module = getModule(reactContext);
+-
+         DownloadManager.Request request = new DownloadManager.Request(Uri.parse(url));
++        AlertDialog.Builder builder = new AlertDialog.Builder(mReactContext);
```
Comment (Cal-L) [inline patches/react-native-webview+11.13.0.patch:236]: Just tagging this line as actual code that changed

**Review by Cal-L — COMMENTED** [review]
Left some comments. You can ignore the areas where I commented that I tagged. It's just a way for me to identify the areas of code that changed. Btw, it looks like there are a bunch of changes that you didn't actually touch. If you have the chance, I'd recommend reapplying the changes that you made and then re-patching to see if you still get all of the diffs.

File: patches/react-native-webview+11.13.0.patch (hunk 3 above)
Comment (blackdevelopa) [reply patches/react-native-webview+11.13.0.patch]: Crossed checked against main and reverted.

**Review by Cal-L — APPROVED** [review]
LGTM!

**Review by cortisiko — CHANGES_REQUESTED** [review]
@Cal-L @blackdevelopa @NicholasEllul This PR does what it was set out to do. I went ahead and did the browser regression and reported my findings [here](https://github.com/MetaMask/mobile-planning/issues/458#issuecomment-1325810589).

**Comment by blackdevelopa** [comment]
@cortisiko I have now updated the cirt app description [here](https://github.com/MetaMask/mobile-planning/issues/458#issuecomment-1326138612)

**Review by cortisiko — APPROVED** [review]
After discussing the above with @blackdevelopa @NicholasEllul we decided that the above is not a blocker for this PR. We will follow up in a different PR. You can track the progress [here](https://github.com/MetaMask/mobile-planning/issues/494)

**Pull request merged**